torch/testing/_internal/common_utils.py

```diff
@@ -2813,7 +2813,10 @@ def _to_number(self, number_like, *, id):
         elif isinstance(number_like, Enum):
             return int(number_like)  # type: ignore[call-overload]
         else:
-            return super()._to_number(number_like, id=id)
+            number = super()._to_number(number_like, id=id)
+            if type(number) not in self._TYPE_TO_DTYPE.keys():
```

Comment:
If i'm not mistaken, one can override type, but not instance
```suggestion
            if not isinstance(number, self._TYPE_TO_DTYPE.keys()):
```

Reply:
As we talked offline. type() and isinstance are different.

if you use isinstance(exampleEnum, int), it will return true. However, if you use type(exampleEnum) == int, it will return false. 

type only returns the type of an object (its class). isinstance checks if a given object (first parameter) is:
1. an instance of a class specified as a second parameter. For example, is variable an instance of the str class?
2. or an instance of a subclass of a class specified as a second parameter. 
https://docs.python.org/3/library/functions.html#type
https://docs.python.org/3/library/functions.html#isinstance